<?xml version="1.0" ?>
<launch xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:noNamespaceSchemaLocation="https://gist.githubusercontent.com/nalt/dfa2abc9d2e3ae4feb82ca5608090387/raw/roslaunch.xsd">
    <!-- Arguments -->
    <arg name="execute_on_niryo_robot_raspberry_pi_image" default="true" />
    <arg name="simulation_mode" default="true"/>
    <arg name="simu_gripper" default="false"/>
    <arg name="gazebo" default="false"/>
    <arg name="can_enabled" default="true"/>
    <arg name="ttl_enabled" default="true"/>
    <arg name="hardware_version" default="ned"/>
    <arg name="debug" default="false"/>
    <arg name="image_version" default="" />

    <group ns="niryo_robot">
        <group ns="info">
            <param name="image_version"     type="string" value="$(arg image_version)"/>
            <param name="ros_version"       type="string" command="rosversion niryo_robot_bringup" />
            <param name="niryo_robot_type"  type="string" value="NIRYO_NED" />
        </group>
    </group>

    <node pkg="niryo_robot_hardware_interface" type="niryo_robot_hardware_interface_node"
          name="niryo_robot_hardware_interface" output="screen">

        <param name="hardware_version" type="string" value="$(arg hardware_version)"/>

        <rosparam file="$(find niryo_robot_hardware_interface)/config/default.yaml" command="load"/>
        <rosparam file="$(find cpu_interface)/config/default.yaml" command="load"/>

        <rosparam file="$(find ttl_driver)/config/$(arg hardware_version)/bus_config.yaml" command="load" ns="ttl_driver"/>
        <rosparam file="$(find ttl_driver)/config/$(arg hardware_version)/dynamixels_params.yaml" command="load" />
        <rosparam file="$(find ttl_driver)/config/default.yaml" command="load"  ns="ttl_driver"/>

        <rosparam file="$(find can_driver)/config/bus_config.yaml" command="load" ns="can_driver"/>
        <rosparam file="$(find can_driver)/config/steppers_params.yaml" command="load" />
        <rosparam file="$(find can_driver)/config/default.yaml" command="load" ns="can_driver"/>

        <rosparam file="$(find joints_interface)/config/$(arg hardware_version)/motors_param.yaml" command="load"/>
        <rosparam file="$(find joints_interface)/config/$(arg hardware_version)/joints_params.yaml" command="load" />
        <rosparam file="$(find joints_interface)/config/default.yaml" command="load" />

        <rosparam file="$(find tools_interface)/config/default.yaml" command="load" ns="tools_interface"/>
        <rosparam file="$(find tools_interface)/config/$(arg hardware_version)/tools_config.yaml" command="load" ns="tools_interface"/>

        <rosparam file="$(find conveyor_interface)/config/default.yaml" command="load" ns="conveyor"/>

        <param name="simulation_mode" type="bool" value="$(arg simulation_mode)"/>
        <param name="gazebo" type="bool" value="$(arg gazebo)"/>
        <param name="simu_gripper" type="bool" value="$(arg simu_gripper)"/>
        <param name="can_enabled" type="bool" value="$(arg can_enabled)"/>
        <param name="ttl_enabled" type="bool" value="$(arg ttl_enabled)"/>
        <param name="debug" type="bool" value="$(arg debug)"/>
    </node>

</launch>
